labelledslider.cpp

```diff
@@ -36,13 +36,17 @@ int LabelledSlider::value() const {
 }
 
 // public slot
-void LabelledSlider::setValue(int v) {
-  slider->setValue(v);
+void LabelledSlider::setValue(double v) {
+  preciseValue = v;
+  slider->setValue((int)(v + 0.5));
 }
 
 // public slot
-void LabelledSlider::changeValue(int v) {
-  slider->setValue(slider->value() + v);
+void LabelledSlider::changeValue(double v) {
+  preciseValue += v;
+  if (preciseValue < slider->minimum()) preciseValue = slider->minimum();
```

Comment:
maybe we should move that range check to `LabeledSlider::setValue(double v)`
otherwise it would be possible to write wrong values to preciseValue with that mehtod

Reply:
Sounds fine to me, I'll change it to do that if you like. But `LabeledSlider::setValue` didn't have that check before, however, and arguably it doesn't need it because it's just calling setValue on the slider and QSliders inherently limit their values to the minimum and maximum. https://doc.qt.io/qt-5/qabstractslider.html#value-prop

Reply:
Scratch that, I see what you mean. We have to set `preciseValue` there. I'll rework it.

Reply:
Fixed.